insights/client/client.py

```diff
@@ -311,36 +310,17 @@ def get_connection(config):
 
 def _legacy_upload(config, pconn, tar_file, content_type, collection_duration=None):
     logger.info('Uploading Insights data.')
-    api_response = None
     for tries in range(config.retries):
-        upload = pconn.upload_archive(tar_file, '', collection_duration)
-
-        if upload.status_code in (200, 201):
-            api_response = json.loads(upload.text)
-
-            # Write to last upload file
-            with open(constants.last_upload_results_file, 'w') as handler:
-                if six.PY3:
-                    handler.write(upload.text)
-                else:
-                    handler.write(upload.text.encode('utf-8'))
-            write_to_disk(constants.lastupload_file)
-
-            msg_name = determine_hostname(config.display_name)
-            account_number = config.account_number
-            if account_number:
-                logger.info("Successfully uploaded report from %s to account %s.",
-                            msg_name, account_number)
-            else:
-                logger.info("Successfully uploaded report for %s.", msg_name)
-            if config.register:
-                # direct to console after register + upload
-                logger.info('View the Red Hat Insights console at https://cloud.redhat.com/insights/')
-            break
-
-        elif upload.status_code in (412, 413):
-            pconn.handle_fail_rcs(upload)
+        try:
+            upload = pconn.upload_archive(tar_file, '', collection_duration)
+        except TimeoutException:
+            # allow timeouts here because we want to retry the upload
+            upload = None
+        except (PayloadTooLargeException, UnregisteredException):
             raise RuntimeError('Upload failed.')
+
+        if upload:
+            break
```

Comment:
Just thinking whether to keep the _else_ branch here. The _break_ interrupts the cycle. But the else on the other hand provides a nice symmetry.